oak_functions/loader/src/server.rs

```diff
@@ -332,16 +328,15 @@ impl WasmState {
                 OakStatus::ErrInvalidArgs
             })?;
 
-        let extension = self
-            .extensions_indices
-            .values_mut()
-            .find(|extension| extension.get_handle() == handle)
-            .ok_or({
-                // TODO(#2715): Add logging, which will be easier when we have a lookup to
-                // extension_indices.
-                // self.log_error(&format!("Cannot find extension with handle {:?}.", handle));
-                OakStatus::ErrInvalidHandle
-            })?;
+        // TODO(mschett): Make more idiomatic
```

Comment:
I should be able to make this more idiomatic, but I cannot see how. The problem here is that `self.log_error` needs a mutable reference to self.

Can you see a way?

Reply:
Can that be made to take &self instead? Either way, what do you mean with more idiomatic? Like

```rust
let extension = self.extensions.get_mut(&handle).or(Err(OakStatus::ErrInvalidHandle))?;
```

but with logging?

Reply:
Unrelated to the borrowing: if anything should happen only when the value is `None` (e.g. logging) it should use `ok_or_else`. With `ok_or` the logging will always happen because the block will first be evaluated and then the result will be passed to `ok_or`.

Regarding the borrowing, I think the only way around it is to get a clone of the logger from self beforehand and then use that logger in the closure to do the logging. I don't think that that would be an improvement over the current code though. One minor thing to make this slightly more idiomatic would be to remove the `return` and `;` in the `None` arm of the match.

Reply:
Thank you, Conrad! Yes, I understand now!